src/lib/rtc-to-net/rtc-to-net.ts

```diff
@@ -344,24 +344,54 @@ import ProxyConfig = require('./proxyconfig');
           // TODO: Add a unit test for this case.
           this.replyToPeer_(socks.Reply.UNSUPPORTED_COMMAND);
           return Promise.reject(e);
-        })
-        .then(this.getTcpConnection_)
-        .then((tcpConnection) => {
-          this.tcpConnection_ = tcpConnection;
-
+        }).then((webEndpoint :net.Endpoint) :Promise<tcp.Connection> => {
+          this.webEndpoint_ = webEndpoint;
+          // Returns socks reply and bound endpoint of connection
+          if (!this.isWebEndpointValid_(webEndpoint)) {
+            this.replyToPeer_(socks.Reply.NOT_ALLOWED);
+            return Promise.reject(new Error(
+                'tried to connect to disallowed address: ' +
+                webEndpoint.address));
+          }
+          log.debug('%1: Creating tcp connection with reproxy settings: %2',
+                    [this.longId(), this.proxyConfig_.reproxy]);
+
+          // Connect to web endpoint directly or through socks proxy
+          if (typeof this.proxyConfig_.reproxy !== 'undefined' &&
+              this.proxyConfig_.reproxy.enabled) {
+            return Promise.resolve(this.getTcpConnection_(
```

Comment:
Cool. You don't need the `Promise.resolve` - if you return a plain object, the next `then()` handler will receive it. So I think you could just drop `getTcpConnection_ ` altogether.